src/plugins/data/common/search/session/types.ts

```diff
@@ -87,10 +89,20 @@ export interface SearchSessionRequestInfo {
   error?: string;
 }
 
-export interface SearchSessionFindOptions {
-  page?: number;
-  perPage?: number;
-  sortField?: string;
-  sortOrder?: string;
-  filter?: string;
+/**
+ * On-the-fly calculated search session status
+ */
+export interface SearchSessionStatusResponse {
+  status: SearchSessionStatus;
+}
+
+/**
+ * List of search session objects with on-the-fly calculated search session statuses
+ */
+export interface SearchSessionsFindResponse
+  extends SavedObjectsFindResponse<SearchSessionSavedObjectAttributes> {
+  /**
+   * Map containing calculated statuses of search sessions from the find response
+   */
+  statuses: Record<string, SearchSessionStatusResponse>;
```

Comment:
Good suggestion 👍 

I'll take a look if it is worth update, 
I think I didn't go this route initially because we use `SavedObjectFindResponse` with a bunch of other properties, and instead of just extending it, I'd had to deep update it